optuna/samplers/random.py

```diff
@@ -84,15 +84,23 @@ def sample_independent(self, study, trial, param_name, param_distribution):
             v = s * param_distribution.step + param_distribution.low
             return int(v)
         elif isinstance(param_distribution, distributions.IntLogUniformDistribution):
+            # adjust highest value
+            high = int(
+                numpy.floor(
+                    (param_distribution.high - param_distribution.low) / param_distribution.step
+                )
+                * param_distribution.step
+                + param_distribution.low
+            )
```

Comment:
(This is a reply to https://github.com/optuna/optuna/pull/1201#discussion_r427713669.)

Thank you for your swift action!

I have two additional comments about this logic.

1. Instead of adjusting the high value in the sampler, how about changing it in the `Trial.suggest_int` similarly to the `Trial.suggest_discrete_uniform`? It calls `_adjust_discrete_uniform_high` in `optuna/trial/_util.py`, and I think we can employ this approach.

2. I think implement this logic without `numpy` as follows:

`(param_distribution.high - param_distribution.low) // param_distribution.step * param_distribution.step + param_distribution.low`

Does it have problems?

Reply:
Thank you for your comments!

**Comment. 1**: Great! It should be done in a similar way to`Trial.suggest_discrete_uniform`.

**Comment. 2**: I think they output different results. 

```python
import numpy

low = 2
high = 10
step = 2
samples = []
log_low = numpy.log(low - 0.5)
log_high = numpy.log(high + 0.5)

# current logic
rnd = numpy.random.RandomState(8)
for _ in range(1000):
    s = numpy.exp(rnd.uniform(log_low, log_high))
    v = (
        numpy.round((s - low) / step)
        * step
        + low
    )
    samples.append(int(min(max(v, low), high)))

# print(numpy.bincount(samples))
# array([  0,   0, 345,   0, 275,   0, 183,   0, 116,   0,  81])

# new logic without numpy
rnd = numpy.random.RandomState(8)
for _ in range(1000):
    s = numpy.exp(rnd.uniform(log_low, log_high))
    v = (
        (s - low) // step
        * step
        + low
    )
    samples.append(int(min(max(v, low), high)))

# print(numpy.bincount(samples))
# array([  0,   0, 842,   0, 502,   0, 321,   0, 224,   0, 111])
```

Reply:
Again,  I misunderstood your comment. I'm sorry if my previous comment confused you.

Reply:
Thank you for your update. The commit https://github.com/optuna/optuna/pull/1201/commits/ad5d3faf02da2a0fed6aac25fc2bb3b8a79bdbf2 seems good to me.